src/asm/asmy.y

```diff
@@ -94,8 +98,25 @@ size_t symvaluetostring(char *dest, size_t maxLength, char *sym)
 
 	} else {
 		uint32_t value = sym_GetConstantValue(sym);
-		int32_t fullLength = snprintf(dest, maxLength + 1, "$%X",
-					      value);
+		int32_t fullLength;
+
+		/* Special cheat for binary */
+		if(mode && !mode[0]) {
+			char binary[32 + 1];
+			char *write_ptr = binary + 32;
+			fullLength = 0;
+			*write_ptr = 0;
+			do {
+				*(--write_ptr) = (value & 1) + '0';
+				value >>= 1;
+				fullLength++;
+			} while(value);
+			strncpy(dest, write_ptr, maxLength + 1);
```

Comment:
I think you have an unterminated string in `binary` here.

Reply:
Except not because the string terminator is written on line 108

Reply:
Ah, right.